pkg/controllers/awsloadbalancercontroller/rbac_clusterrolebinding.go

```diff
@@ -0,0 +1,138 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+func (r *AWSLoadBalancerControllerReconciler) ensureClusterRoleBinding(ctx context.Context, sa *corev1.ServiceAccount, controller *albo.AWSLoadBalancerController) error {
+	reqLogger := log.FromContext(ctx)
+
+	desired := desiredClusterRoleBinding(ctx, sa)
+	reqLogger.Info("ensuring clusterrolebindings", "clusterrolebindings", desired.Name)
+
+	if err := controllerutil.SetControllerReference(controller, desired, r.Scheme); err != nil {
+		return fmt.Errorf("failed to set the controller reference for clusterrolebindings %s : %w", desired.Name, err)
+	}
+
+	exist, current, err := r.currentClusterRoleBinding(ctx, desired)
+	if err != nil {
+		return fmt.Errorf("failed to fetch current clusterrolebindings: %w", err)
+	}
+
+	if !exist {
+		if err := r.createClusterRoleBinding(ctx, desired); err != nil {
+			return err
+		}
+
+		_, current, err = r.currentClusterRoleBinding(ctx, desired)
+		if err != nil {
+			return fmt.Errorf("failed to fetch created clusterrolebindings: %w", err)
+		}
+		reqLogger.Info("created clusterrolebindings", "clusterrolebindings", desired.Name)
+	}
+
+	if err := r.updateClusterRoleBinding(ctx, current, desired); err != nil {
+		return fmt.Errorf("failed to update clusterrolebindings: %w", err)
+	}
+
+	return nil
+}
+
+// updateClusterRoleBindings updates the current clusterrolebindings and returns a flag to denote if the update was done.
+func (r *AWSLoadBalancerControllerReconciler) updateClusterRoleBinding(ctx context.Context, current *rbacv1.ClusterRoleBinding, desired *rbacv1.ClusterRoleBinding) error {
+	changed, err := isClusterRoleBindingChanged(current, desired)
+	if err != nil {
+		return err
+	}
+
+	if !changed {
+		return nil
+	}
+
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
+	updated.Namespace = desired.Namespace
+	updated.Subjects = desired.Subjects
+	updated.RoleRef = desired.RoleRef
+
+	if err := r.Client.Update(ctx, updated); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isClusterRoleBindingChanged(current *rbacv1.ClusterRoleBinding, desired *rbacv1.ClusterRoleBinding) (bool, error) {
+	if current.Name != desired.Name {
+		return true, nil
+	}
+
+	if current.Namespace != desired.Namespace {
+		return true, nil
+	}
+
+	if !(reflect.DeepEqual(current.Subjects, desired.Subjects)) || !(reflect.DeepEqual(current.RoleRef, desired.RoleRef)) {
```

Comment:
I'm afraid this will always fail because `Subject` has `APIGroup` field which has defaults which is not set in the desired.

Reply:
Actually, not it will remain empty. So this is ok.

Reply:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#subject-v1-rbac-authorization-k8s-io

Reply:
It defaults to "" for type service accounts.